sign_language_datasets/datasets/dgs_corpus/dgs_corpus.py

```diff
@@ -222,75 +287,127 @@ def _split_generators(self, dl_manager: tfds.download.DownloadManager):
 
         if self._builder_config.split is not None:
             split = load_split(self._builder_config.split)
 
-            train_data = {key: value for key, value in processed_data.items() if key in split["train"]}
-            dev_data = {key: value for key, value in processed_data.items() if key in split["dev"]}
-            test_data = {key: value for key, value in processed_data.items() if key in split["test"]}
+            train_args = {"data": data, "split": split["train"]}
+            dev_args = {"data": data, "split": split["dev"]}
+            test_args = {"data": data, "split": split["test"]}
 
             return [
-                tfds.core.SplitGenerator(name=tfds.Split.TRAIN, gen_kwargs={"data": train_data}),
-                tfds.core.SplitGenerator(name=tfds.Split.VALIDATION, gen_kwargs={"data": dev_data}),
-                tfds.core.SplitGenerator(name=tfds.Split.TEST, gen_kwargs={"data": test_data}),
+                tfds.core.SplitGenerator(name=tfds.Split.TRAIN, gen_kwargs=train_args),
+                tfds.core.SplitGenerator(name=tfds.Split.VALIDATION, gen_kwargs=dev_args),
+                tfds.core.SplitGenerator(name=tfds.Split.TEST, gen_kwargs=test_args),
             ]
 
         else:
-            return [tfds.core.SplitGenerator(name=tfds.Split.TRAIN, gen_kwargs={"data": processed_data})]
+            return [tfds.core.SplitGenerator(name=tfds.Split.TRAIN, gen_kwargs={"data": data})]
+
+    def _include_videos(self, datum: Any, features: Any):
+        videos = {}
+        for participant in ["a", "b", "c"]:
+            video_key = "video_" + participant
+
+            if video_key not in datum.keys() or datum[video_key] is None:
+                video = ""
+            else:
+                video = str(datum[video_key])
+
+            videos[participant] = video
+
+        # make sure that the video fps is as expected
+        for video_path in videos.values():
+            if video_path == "":
+                continue
 
-    def _generate_examples(self, data):
+            cap = cv2.VideoCapture(video_path)
+            actual_video_fps = cap.get(cv2.CAP_PROP_FPS)
+            cap.release()
+
+            assert math.isclose(actual_video_fps, float(DEFAULT_FPS)), "Framerate of video '%s' is %f instead of %d" % (
+                video_path,
+                actual_video_fps,
+                DEFAULT_FPS,
+            )
+
+        features["fps"] = self._builder_config.fps if self._builder_config.fps is not None else DEFAULT_FPS
+        features["paths"]["videos"] = videos
+
+    def _generate_examples(self, data, split: List[str] | Dict[str, List[str]] = None):
         """ Yields examples. """
 
-        default_fps = 50
         default_video = np.zeros((0, 0, 0, 3))  # Empty video
 
         for _id, datum in list(data.items()):
+            if split is not None and _id not in split:
```

Comment:
If for sentence-level, ids are now `f'{features["id"]}_{sentence["id"]}' ` then known split files would need to a) also have this structure or b) be preprocessed slightly when they are loaded. Which of those would you prefer? (or, of course, a third option)

Reply:
How about I just add a `document_id` field?

Reply:
that would also work; still the code here to compare ids to a split would need to be tweaked slightly because you said your preferred format for document splits would be something like

```json
{
    "document_id": [
          "sentence_id_1", ...
          ],
          ...
}
```